test/test_common_url_handling.rb

```diff
@@ -0,0 +1,160 @@
+require_relative 'test_helper'
+
+# URLHandling tests
+class TestCommonUrlHandling < MiniTest::Test
+  def test_no_path_in_uri
+    client = Kubeclient::Client.new('http://localhost:8080', 'v1')
+    rest_client = client.rest_client
+    assert_equal('v1', client.instance_variable_get(:@api_version))
+    assert_equal('', client.instance_variable_get(:@api_group))
+    assert_equal('http://localhost:8080/api/v1', rest_client.url.to_s)
+  end
+
+  def test_with_api_path_in_uri
+    client = Kubeclient::Client.new('http://localhost:8080/api', 'v1')
+    rest_client = client.rest_client
+    assert_equal('v1', client.instance_variable_get(:@api_version))
+    assert_equal('', client.instance_variable_get(:@api_group))
+    assert_equal('http://localhost:8080/api/v1', rest_client.url.to_s)
+  end
+
+  def test_with_api_path_in_uri_other_version
+    client = Kubeclient::Client.new('http://localhost:8080/api', 'v2')
+    rest_client = client.rest_client
+    assert_equal('v2', client.instance_variable_get(:@api_version))
+    assert_equal('', client.instance_variable_get(:@api_group))
+    assert_equal('http://localhost:8080/api/v2', rest_client.url.to_s)
+  end
+
+  def test_with_api_group_path_in_uri
+    client = Kubeclient::Client.new('http://localhost:8080/apis/this_is_the_group', 'v1')
+    rest_client = client.rest_client
+    assert_equal('v1', client.instance_variable_get(:@api_version))
+    assert_equal('this_is_the_group/', client.instance_variable_get(:@api_group))
```

Comment:
The trailing slash looked surprising to me, but yep, we had it before and we're sending
`hash[:apiVersion] = @api_group + @api_version`
so this is good :heavy_check_mark:.
(Also this representation is convenient for automatically handling both `foo/v1` and core `v1`)